### PR TITLE
[WFLY-12945] Refactoring undertow subsystem to use new MSC value API

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFLY-12945

### CODE REVIEW

**Comment by darranl** [comment]
Just placing on hold for now, we have run into some error reporting issues with the new approach that may need looking into first.

**Comment by ropalka** [comment]
Retest this please.

**Comment by ropalka** [comment]
Please remove the hold flag @darranl the conflict have been fixed.

**Comment by darranl** [comment]
@ropalka Sorry didn't see the other comment - FYI the hold label was not in relation to the conflicts it was added as we were running into issues over on WildFly Core where use of the new MSC API was dropping errors starting the server so was leading to failed start ups with no error reporting as to what the underlying problem was.

We were able to manually debug and fix the underlying issue but it was still a concern that the error to diagnose the error was no longer reported.

**Comment by ropalka** [comment]
Could I reproduce it somehow @darranl ? I can't see any reported issue in MSC JIRA.

**Comment by darranl** [comment]
@ropalka I think we can give this another go but for now I think avoid any changes to the JACC related services, those are already in a bit of complicated position as we need to backport some of them.

**Comment by ropalka** [comment]
Are you aware of this PR @bstansberry @kabir ?

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpListenerService.java [hunk 1]
```diff
@@ -54,24 +55,21 @@
...
  */
 public class HttpListenerService extends ListenerService {
     private volatile AcceptingChannel<StreamConnection> server;
 
     private final ChannelUpgradeHandler httpUpgradeHandler = new ChannelUpgradeHandler();
-    protected final InjectedValue<ListenerRegistry> httpListenerRegistry = new InjectedValue<>();
-    /**
-     * @deprecated Replaced by HTTP_UPGRADE_REGISTRY.getCapabilityServiceName()
-     */
-    @Deprecated
```
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpListenerService.java:66]: This comment and annotation should not have been removed.

File: undertow/src/main/java/org/wildfly/extension/undertow/Host.java [hunk 2]
```diff
@@ -132,13 +139,13 @@ private String getDeployedContextPath(DeploymentInfo deploymentInfo) {
 
     @Override
     public void start(StartContext context) throws StartException {
-        suspendControllerInjectedValue.getValue().registerActivity(suspendListener);
-        if(suspendControllerInjectedValue.getValue().getState() == SuspendController.State.RUNNING) {
+        suspendController.get().registerActivity(suspendListener);
+        if(suspendController.get().getState() == SuspendController.State.RUNNING) {
```
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/Host.java:143]: @jasondlee fyi ^^^ is going to result in conflicts

Comment (jasondlee) [reply undertow/src/main/java/org/wildfly/extension/undertow/Host.java:143]: Noted. I'll keep an eye out so I can merge and resolve as soon as I can.

**Comment by bstansberry** [comment]
@ropalka Can you fix the conflict and we'll see if we can get this done in the next few days? Thanks.

**Comment by ropalka** [comment]
Fixed @bstansberry including your feedback about @Deprecated annotation

**Comment by darranl** [comment]
Sorry @ropalka but looks like a bunch of conflicts again, I think we should try again as we are at the start of WildFly 24 development.

**Comment by ropalka** [comment]
Fixed @darranl

**Comment by darranl** [comment]
Thanks @ropalka now is the time to get this in before any further refactoring of the Undertow subsystem begins for WildFly 24

**Comment by bstansberry** [comment]
Thanks, @ropalka -- sorry this took so long.

**Pull request merged**